komodo/check_up_to_date_pypi.py

```diff
@@ -158,19 +162,47 @@ def run_check_up_to_date(
                 "Writing upgrade proposals from pypi, "
                 "assuming nothing has changed with dependencies..."
             )
-            with open(propose_upgrade, mode="w", encoding="utf-8") as fout:
+            with open(propose_upgrade, "w") as fout:
                 yaml.dump(releases, fout)
             write_to_file(repository, repository_file)
 
-        errors = []
+        major_upgrades, minor_upgrades, patch_upgrades, other_upgrades = [], [], [], []
         for name, versions in upgrade_proposals_from_pypi.items():
-            pypi_latest = versions["suggested"]
-            current_version = versions["previous"]
-            errors.append(
-                f"{name} not at latest pypi version: {pypi_latest}, "
-                f"is at: {current_version}"
+            pypi_latest = get_version.Version(versions["suggested"])
+            current_version = get_version.Version(versions["previous"])
+            if pypi_latest.major > current_version.major:
+                major_upgrades.append(
+                    f"{name} not at latest pypi version: {pypi_latest}, "
+                    f"is at: {current_version}"
+                )
+            elif pypi_latest.minor > current_version.minor:
+                minor_upgrades.append(
+                    f"{name} not at latest pypi version: {pypi_latest}, "
+                    f"is at: {current_version}"
+                )
+            elif pypi_latest.micro > current_version.micro:
+                patch_upgrades.append(
+                    f"{name} not at latest pypi version: {pypi_latest}, "
+                    f"is at: {current_version}"
+                )
+            else:
+                other_upgrades.append(
+                    f"{name} not at latest pypi version: {pypi_latest}, "
+                    f"is at: {current_version}"
+                )
+        sys.exit(
```

Comment:
Nonzero argument to `sys.exi`t usually means something bad happened. I would prefer `print` statements, and then followed by `sys.exit(someinteger)`. Then we can discuss whether upgrades being present is something that should give a nonzero exit value. Do we want some script (e.g. a github workflow) to stop if there are upgrades present? If so, that would argue for a nonzero exit value.

Reply:
(changing from sys.exit to print is really unrelated to the issue you are solving, so you could prepend your PR with a commit that first changes from sys.exit to print)